mesonbuild/interpreter/type_checking.py

```diff
@@ -530,9 +530,24 @@ def link_whole_validator(values: T.List[T.Union[StaticLibrary, CustomTarget, Cus
     NATIVE_KW,
 ]
 
+
+def _name_validator(arg: T.Optional[T.Union[str, T.List]]) -> T.Optional[str]:
+    if isinstance(arg, list) and arg:
+        return 'may be an empty array to signify default.'
+    return None
+
+
+_NAME_PREFIX_KW: KwargInfo[T.Optional[str]] = KwargInfo(
+    'name_prefix',
+    (str, NoneType, ContainerTypeInfo(list, object)),
```

Comment:
Since we have a custom validator to verify what kind of list we get here, I think we probably do not need to go through a ContainerTypeInfo for it? Wouldn't it work equally well to do:
```python
(str, NoneType, list)
```